src/main/java/io/strimzi/kafka/quotas/AvailableBytesThrottleFactorPolicy.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.kafka.quotas;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.slf4j.Logger;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Determines if the number of available bytes on any given volume falls below the configured limit.
+ */
+public class AvailableBytesThrottleFactorPolicy implements ThrottleFactorPolicy {
+    private final Logger log = getLogger(AvailableBytesThrottleFactorPolicy.class);
+    private final List<Runnable> listeners;
+    private final long availableBytesLimit;
+
+    private volatile boolean throttled = false;
+
+    /**
+     * Creates and configures the throttle factor supplier
+     * @param availableBytesLimit the minimum number of bytes below which the throttle should be applied.
+     */
+    public AvailableBytesThrottleFactorPolicy(long availableBytesLimit) {
+        this.availableBytesLimit = availableBytesLimit;
+        listeners = new CopyOnWriteArrayList<>();
+    }
+
+    /**
+     * Accepts an updated collection of volumes to validate against the configured limit.
+     * @param volumes the new collection of volumes to be considered
+     */
+    @Override
+    public void observeVolumeUsage(Collection<VolumeUsage> volumes) {
+        boolean initial = throttled;
+        throttled = calculateNewFactor(volumes);
+        if (throttled != initial) {
+            log.debug("throttle status changed from: {} to {} on update", initial, throttled);
+            listeners.forEach(Runnable::run);
+        }
+    }
+
+    private boolean calculateNewFactor(Collection<VolumeUsage> volumes) {
```

Comment:
This doesn't actually calculate a factor. Maybe call it `shouldThrottle`?

Reply:
good point resolved with calculateNewFactor no longer calculates so give it a better name :D Sam Barker A minute ago